### PR TITLE
Improving uft test discovery and execution

### PR DESCRIPTION
**Pull request by radislavB**
1. Support discovery of data tables
2. Improving JobCleaner to remove orphan discovery jobs
3.Support "executable" field of automated tests : Set discovered test  as executable and set deleted tests as not-executable 

4.fix running test from file system in mtbx file on slave
5.Fix support of folder plugin

https://issues.jenkins-ci.org/browse/JENKINS-44392

### CODE REVIEW

**Comment by YafimK** [comment]
- [ ] will need to be verified by UFT and LR due to change in common file in a critical section.

**Comment by m-seldin** [comment]
approved for test build

**Comment by YafimK** [comment]
approved for test build

File: src/main/java/com/hp/application/automation/tools/run/RunFromFileBuilder.java [hunk 1]
```diff
@@ -497,11 +504,10 @@ private String createMtbxFileInWs(FilePath workspace, String mtbxContent) {
 			InputStream in = IOUtils.toInputStream(mtbxContentUpdated, "UTF-8");
 			remoteFile.copyFrom(in);
 
-			String filePath = remoteFile.getRemote();
-			return filePath;
+			return remoteFile.getRemote();
 
 		} catch (IOException | InterruptedException e) {
-			throw new RuntimeException("Failed to save MTBX file : " + e.getMessage());
+			throw new RuntimeException("Failed to save MTBX file : " + e.getMessage(), e);
```
Comment (YafimK) [inline src/main/java/com/hp/application/automation/tools/run/RunFromFileBuilder.java]: Better use checked exception, log and fail the build properly then crash it with RuntimeException.
Catch in the caller and use - i.e. something like:
listener.error("Cool msg on awful error " + e);
build.setResult(Result.FAILURE);

@m-seldin @radislavB WDYT?

Comment (radislavB) [reply src/main/java/com/hp/application/automation/tools/run/RunFromFileBuilder.java]: @YafimK done

**Comment by YafimK** [comment]
Approved for test build

**Pull request merged**